locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/BasicLocationNetworkInfoCustomizer.java

```diff
@@ -0,0 +1,472 @@
+
+    // --------------------------------------------------------------------------------------
+
+    public BasicLocationNetworkInfoCustomizer() {
+        super();
+    }
+
+    public BasicLocationNetworkInfoCustomizer(Map<?, ?> params) {
+        super(params);
+    }
+
+    public BasicLocationNetworkInfoCustomizer(final ConfigBag params) {
+        super(params);
+    }
+
+    // --------------------------------------------------------------------------------------
+
+    /**
+     * Overrides the behaviour of {@link BasicJcloudsLocationCustomizer#apply(EntityLocal)} to set
+     * the instance as the value of {@link JcloudsLocationConfig#LOCATION_NETWORK_INFO_CUSTOMIZER},
+     * rather than in its provisioning properties.
+     */
+    @Override
+    public void apply(EntityLocal entity) {
+        ConfigKey<Object> subkey = BrooklynConfigKeys.PROVISIONING_PROPERTIES.subKey(JcloudsLocationConfig.LOCATION_NETWORK_INFO_CUSTOMIZER.getName());
+        entity.config().set(subkey, this);
+        LOG.debug("{} set itself as the location network info customizer on {}", this, entity);
+    }
+
+    // --------------------------------------------------------------------------------------
+
+    /**
+     * Combines the given resolve options with the customiser's configuration to determine the
+     * best address and credential pair for management. In particular, if the resolve options
+     * allow it will check that the credential is actually valid for the address.
+     */
+    @Override
+    public ManagementAddressResolveResult resolve(
+            JcloudsLocation location, NodeMetadata node, ConfigBag config, ManagementAddressResolveOptions options) {
+        LOG.debug("{} resolving management parameters for {}, node={}, config={}, options={}",
+                new Object[]{this, location, node, config, options});
+        Stopwatch timer = Stopwatch.createStarted();
+        // Should only be null in tests.
+        final Entity contextEntity = getContextEntity(config);
+        if (shouldPublishNetworks() && options.publishNetworkSensors() && contextEntity != null) {
+            publishNetworks(node, contextEntity);
+        }
+        HostAndPort hapChoice = null;
+        LoginCredentials credChoice = null;
+
+        Iterable<HostAndPort> managementCandidates = getManagementCandidates(location, node, config, options);
+        Iterable<LoginCredentials> credentialCandidates = getCredentialCandidates(location, node, options, config);
+
+        // Try each pair of address and credential until one succeeds.
+        if (options.expectReachable() && options.pollForFirstReachableAddress() && shouldCheckCredentials()) {
+            for (HostAndPort hap : managementCandidates) {
+                for (LoginCredentials cred : credentialCandidates) {
+                    LOG.trace("Testing host={} with credential={}", hap, cred);
+                    if (checkCredential(location, hap, cred, config, options.isWindows())) {
```

Comment:
The check is still vulnerable to the local network IP poisoning. The reachability check will succeed immediately for the local IP, not giving enough time for the public IP to become available. And then the credential check will fail.